cartography/intel/semgrep/findings.py

```diff
@@ -110,48 +143,64 @@ def transform_sca_vulns(raw_vulns: List[Dict[str, Any]]) -> Tuple[List[Dict[str,
+            sca_vuln["ruleId"] = rule_id
+            sca_vuln["title"] = package + ":" + vulnerability_class
+            sca_vuln["description"] = vuln["rule"]["message"]
+            sca_vuln["ecosystem"] = vuln["found_dependency"]["ecosystem"]
+            sca_vuln["severity"] = vuln["severity"].upper()
+            sca_vuln["reachability"] = vuln["reachability"].upper()  # Check done to determine rechabilitity
+            sca_vuln["reachableIf"] = vuln["reachable_condition"].upper() if vuln["reachable_condition"] else None
+            sca_vuln["exposureType"] = _determine_exposure(vuln)  # Determintes if reachable or unreachable
+            dependency = f"{package}|{vuln['found_dependency']['version']}"
+            sca_vuln["matchedDependency"] = dependency
+            dep_url = vuln["found_dependency"]["lockfile_line_url"]
+            if dep_url:  # Lock file can be null, need to set
+                dep_file = dep_url.split("/")[-1].split("#")[0]
+                sca_vuln["dependencyFileLocation_path"] = dep_file
+                sca_vuln["dependencyFileLocation_url"] = dep_url
+            else:
+                if sca_vuln.get("location"):
+                    sca_vuln["dependencyFileLocation_path"] = sca_vuln["location"]["file_path"]
+            sca_vuln["transitivity"] = vuln["found_dependency"]["transitivity"].upper()
+            if vuln.get("vulnerability_identifier"):
+                vuln_id = vuln["vulnerability_identifier"].upper()
+                sca_vuln["cveId"] = vuln_id
+                sca_vuln["ref_urls"] = [_build_vuln_url(vuln_id)]
+            if vuln.get('fix_recommendations') and len(vuln['fix_recommendations']) > 0:
+                fix = vuln['fix_recommendations'][0]
+                dep_fix = f"{fix['package']}|{fix['version']}"
+                sca_vuln["closestSafeDependency"] = dep_fix
+            sca_vuln["openedAt"] = vuln["created_at"]
+            sca_vuln["fixStatus"] = vuln["status"]
+            sca_vuln["triageStatus"] = vuln["triage_state"]
+            sca_vuln["confidence"] = vuln["confidence"]
+            usage = vuln.get("usage")
+            if usage:
+                usage_dict = {}
+                url = usage["location"]["url"]
+                usage_dict["SCA_ID"] = sca_vuln["id"]
+                usage_dict["findingId"] = hash(url.split("github.com/")[-1])
+                usage_dict["path"] = usage["location"]["path"]
+                usage_dict["startLine"] = usage["location"]["start_line"]
+                usage_dict["startCol"] = usage["location"]["start_col"]
+                usage_dict["endLine"] = usage["location"]["end_line"]
+                usage_dict["endCol"] = usage["location"]["end_col"]
+                usage_dict["url"] = url
+                usages.append(usage_dict)
+            vulns.append(sca_vuln)
+        except KeyError as e:
+            logger.warning(f"Error transforming Semgrep SCA vuln {vuln}: {e}")
+            continue
```

Comment:
are all the keys required? Why is it a warning log? 

like, if we look at the results in Cartography and it might not be obvious that something might have failed, so how can we get notified and debug?

Reply:
I know, I don't feel great about it but it mitigates the problem where there are data issues coming from the API and it doesn't block the full ingestion for such problems.

What would you suggest to tackle the problem when the API is missing information we have as mandatory to ingest (like reachability, repository or severity)?

Reply:
what about using `error` log instead? So the ingestion doesn't get blocked and we can still get notified?

if there are data that are optional, the decision about what to do them might need to be a little more intentional
e.g. if missing reachability, just default to `unknown` may be?